router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/marketobulkupload.go

```diff
@@ -0,0 +1,325 @@
+	var succesfulJobIDs, failedJobIDsTrans []int64
+	var err error
+	if ok {
+		succesfulJobIDs, err = misc.ConvertStringInterfaceToIntArray(keyMap["successfulJobs"])
+		if err != nil {
+			failedJobIDsTrans = importingJobIDs
+		}
+	}
+	_, ok = keyMap["unsuccessfulJobs"].([]interface{})
+	if ok {
+		failedJobIDsTrans, err = misc.ConvertStringInterfaceToIntArray(keyMap["unsuccessfulJobs"])
+		if err != nil {
+			failedJobIDsTrans = importingJobIDs
+		}
+	}
+	return succesfulJobIDs, failedJobIDsTrans
+}
+
+func (b *MarketoBulkUploader) Upload(destination *backendconfig.DestinationT, asyncDestStruct *common.AsyncDestinationStruct) common.AsyncUploadOutput {
+	resolveURL := func(base, relative string) string {
+		// var logger logger.Logger
+		baseURL, _ := url.Parse(base)
+		// if err != nil {
+		// 	logger.Fatal(err)
+		// }
+		relURL, _ := url.Parse(relative)
+		// if err != nil {
+		// 	logger.Fatal(err)
+		// }
+		destURL := baseURL.ResolveReference(relURL).String()
+		return destURL
+	}
+	destinationID := destination.ID
+	destinationUploadUrl := asyncDestStruct.URL
+	url := resolveURL(b.TransformUrl, destinationUploadUrl)
+	filePath := asyncDestStruct.FileName
+	destConfig := destination.Config
+	destType := destination.DestinationDefinition.Name
+	failedJobIDs := asyncDestStruct.FailedJobIDs
+	importingJobIDs := asyncDestStruct.ImportingJobIDs
+
+	file, err := os.Open(filePath)
+	if err != nil {
+		panic("BRT: Read File Failed" + err.Error())
+	}
+	defer file.Close()
+	var input []common.AsyncJob
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		var tempJob common.AsyncJob
+		jobBytes := scanner.Bytes()
+		err := json.Unmarshal(jobBytes, &tempJob)
+		if err != nil {
+			panic("Unmarshalling a Single Line Failed")
+		}
+		input = append(input, tempJob)
+	}
+	payload, err := json.Marshal(common.AsyncUploadT{
+		Input:    input,
+		Config:   destConfig,
+		DestType: strings.ToLower(destType),
+	})
+	if err != nil {
+		panic("BRT: JSON Marshal Failed " + err.Error())
+	}
+
+	uploadTimeStat := stats.Default.NewTaggedStat("async_upload_time", stats.TimerType, map[string]string{
+		"module":   "batch_router",
+		"destType": destType,
+	})
+
+	payloadSizeStat := stats.Default.NewTaggedStat("payload_size", stats.TimerType, map[string]string{
+		"module":   "batch_router",
+		"destType": destType,
+	})
+
+	startTime := time.Now()
+	payloadSizeStat.Observe(float64(len(payload)))
+	pkgLogger.Debugf("[Async Destination Maanger] File Upload Started for Dest Type %v", destType)
+	responseBody, statusCodeHTTP := misc.HTTPCallWithRetryWithTimeout(url, payload, config.GetDuration("HttpClient.marketoBulkUpload.timeout", 30, time.Second))
+	pkgLogger.Debugf("[Async Destination Maanger] File Upload Finished for Dest Type %v", destType)
+	uploadTimeStat.Since(startTime)
+	var bodyBytes []byte
+	var statusCode string
+	if statusCodeHTTP != 200 {
+		bodyBytes = []byte(`"error" : "HTTP Call to Transformer Returned Non 200"`)
```

Comment:
Store the actual returned http status code. Will be helpful in debugging.

Reply:
done

Reply:
did not understand it